### DISCUSSION TITLE
How to display third party iframe videos

### DISCUSSION THREAD

**Discussion by berkayyildiz** [sachinchoolur/lightGallery#1083] [Q&A] [answered]
Hello, we are using bunny.net as our cdn. For videos we are using their bunny stream service and they have custom player that we can embed with iframe. Is there any native way to add an third party iframe to gallery. If there is not is it better to create a plugin or contribute to existing video plugin?

Example bunny iframe

```
<div style="position: relative; padding-top: 56.25%;">
  <iframe src="https://iframe.mediadelivery.net/embed/MEDIA_URL?autoplay=true"
          loading="lazy"
          style="border: none; position: absolute; top: 0; height: 100%; width: 100%;"
          allow="accelerometer; gyroscope; autoplay; encrypted-media; picture-in-picture;"
          allowfullscreen="true"></iframe>
</div>

```

**Comment by sachinchoolur** [comment]
Hey @berkayyildiz,

Sorry for the delayed response. I missed you message 

You can try using the iframe option. https://www.lightgalleryjs.com/demos/iframe/
If it doesn't work as intended, you can try creating a new video plugin. 
If you could provide a real video link, I can try from my side

**Reply by berkayyildiz** [reply]
It seem like it is not working for me, here there is a real video link

```
https://iframe.mediadelivery.net/embed/5047/3ad083c7-166e-4144-8862-7f6595e8972e?autoplay=true
```

**Reply by berkayyildiz** [reply]
I was experimenting and tried to do a trick with css.

Normally I was getting error on console about fetch
```
DevTools failed to load source map: Could not load content for https://bunnycdn-video-assets.b-cdn.net/plyr.polyfilled.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
blob:https://iframe.mediadelivery.net/e52aaa98-1958-4f76-9294-8e2c1321c5c4:1 GET blob:https://iframe.mediadelivery.net/e52aaa98-1958-4f76-9294-8e2c1321c5c4 net::ERR_FILE_NOT_FOUND
DevTools failed to load source map: Could not load content for https://cdn.jsdelivr.net/npm/hls.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

I checked and saw that there is a problem with positioning. I tried this and it can fetch and show like this. But of course we can't swipe. I don't know how this is related with css but that was the result.
```
.lg-item .lg-has-iframe iframe {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  width: 100%;
  height: 100%;
}
```

**Reply by sachinchoolur** [reply]
I'm getting 403 error when I try to open the video link

**Reply by berkayyildiz** [reply]
Thats not a video link, thats a link for iframe. If you try in an iframe it should work. Sorry I falsely mentioned it as video url.

```
<div style="position: relative; padding-top: 56.25%;"><iframe src="https://iframe.mediadelivery.net/embed/5047/3ad083c7-166e-4144-8862-7f6595e8972e?autoplay=true" loading="lazy" style="border: none; position: absolute; top: 0; height: 100%; width: 100%;" allow="accelerometer; gyroscope; autoplay; encrypted-media; picture-in-picture;" allowfullscreen="true"></iframe></div>
```

**Reply by sachinchoolur** [reply]
Tried the above code. this is what I'm able to see. Is there any kind of authentication required?
<img width="998" alt="Screenshot 2021-07-08 at 12 45 17 PM" src="https://user-images.githubusercontent.com/6586706/124879051-6f71c380-dfea-11eb-9937-3e94881d4aee.png">

**Reply by berkayyildiz** [reply]
There is not any authentication, I have no idea why you get 403. I don't thing that this will make any difference but you can check this [codepen](https://codepen.io/berkay-yildiz/pen/WNjxvjL) here may be it work on codepen. 

Anyway actually even when we add it with iframe without any additional css I think it is not possible to swipe iframes on gallery to go next. item. Is there any solution for that, because on mobile we do not show controls and user stucks on video. Is creating a plugin solves swipe problem or is there a way to swipe iframes?

**Reply by sachinchoolur** [reply]
You can set the width of the iframe  via `iframeWidth` option

https://www.lightgalleryjs.com/docs/settings/#iframeWidth

```js
lightGallery(document.getElementById('container'), {
    iframeWidth: '50%',
})
```

Since the iframe option is not designed for displaying videos, you might not get all desired functionalities. 

Creating s specific plugin would definitely help

**Comment by berkayyildiz** [comment, marked as answer]
We solved this issue with not using the player from bunny. We realized lightgalley has videojs support and videojs has HLS streaming support. When type set as "application/x-mpegURL" and hls playlist url given, videojs can stream without any problem and light gallery works very well with swipe gestures and auto stop playing. At the end bunny player was not the important part, we was trying to just display videos on light gallery but bunny stream was only providing hls streams, so being able to display stream solved our issue.

````
 lightGallery(EL_REF, {
      plugins: [lgVideo],
      videojs: true,
      })
````

````
<GalleryItem>{
          video: {
            source: [
              {
                src: HLS_PLAYLIST_URL,
                type: "application/x-mpegURL",
              },
            ],
            attributes: { preload: false, controls: true },
          },
        }
````
